frontend/src/utils/useURLState.tsx

```diff
@@ -37,9 +41,13 @@ export function useURLState<T>(
         // Convert the object to a string using JSON because we don't know
         // what type of data we're working with in advance. Also helps to
         // cast non string value. For instance, "2" will become the number 2
-        const paramJSONValue = params.get(paramToWatch);
-        if (paramJSONValue !== null) {
-            setParamValue(JSON.parse(paramJSONValue) as T);
+        const urlParamValue = params.get(paramToWatch);
+        if (urlParamValue !== null) {
+            // TS isn't able to see that stateIsString true implies urlParamValue is a string
```

Comment:
I don't understand why it should be too.

Reply:
I would be interested in an explanation
If it works, I think a typeguard would be safer and more explicit: `urlParamValue as string`